stdlib/std.jsonnet

```diff
@@ -206,18 +206,27 @@ limitations under the License.
       {
         indexable: indexable,
         index:
-          if index == null then 0
-          else index,
+          if index == null
+          then 0
+          else
+            if index < 0
+            then std.length(indexable) + index
+            else index,
```

Comment:
This can result in `invar.index` being a negative number, which subsequently causes a runtime error as `invar.index` is used as the starting index for iteration, and (currently) array/string indexing does not accept negative values.

Compare Python and Jsonnet (with this patch):
```
$ ./jsonnet -e '("012345")[-6:]'
"012345"
$ ./jsonnet -e '("012345")[-7:]'
RUNTIME ERROR: string bounds error: -1 not within [0, 6)
$ python3 -c 'print(("012345")[-6:])'
012345
$ python3 -c 'print(("012345")[-7:])'
012345
```

Reply:
Feel free to work on this, might take a while from me to cycle back to this.

Reply:
Done in 422af989b6d63963e4e6ab693091f544a623857b